### PR TITLE
1125-fesso-add-status-dropdown

### PR DESCRIPTION
**Pull request by NataliaR-BCN**

### CODE REVIEW

**Review by marmorrei — APPROVED** [review]

File: web/usuaris/src/__mocks__/handlers.ts [hunk 1]
```diff
@@ -5,11 +5,13 @@ export const handlers = [
   http.get(urls.getUsers, ({ request }) => {
     const url = new URL(request.url)
     const itinerarySlug = url.searchParams.get('itinerarySlug')
+    const status = url.searchParams.get('status')
     const name = url.searchParams.get('name')
     const dni = url.searchParams.get('dni')
 
     if (
       itinerarySlug === 'frontend-react' &&
+      status === 'ACTIVE' &&
```
Comment (kevinmamaqi) [inline web/usuaris/src/__mocks__/handlers.ts:14]: Utilizar ENUM

File: web/usuaris/src/__tests__/molecules/StatusDropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  it('renders all status with icon and returns selected option to parent', async () => {
+    render(<StatusDropdown handleStatus={mockHandleClick} />)
+
+    const dropdownHeader = screen.getByTestId('dropdown-header')
+    expect(dropdownHeader).toHaveTextContent(/estat/i)
+
+    fireEvent.click(dropdownHeader)
+
+    const activeStatus = screen.getByText(/actiu/i)
```
Comment (kevinmamaqi) [inline web/usuaris/src/__tests__/molecules/StatusDropdown.test.tsx:28]: getByTestId

File: web/usuaris/src/__tests__/molecules/StatusDropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    await waitFor(() => expect(screen.getByText(/pendent/i)).toBeVisible())
+
+    const blockedStatus = screen.getByText(/bloquejat/i)
+
+    fireEvent.click(blockedStatus)
+
+    expect(dropdownHeader).toHaveTextContent(/bloquejat/i)
+
+    expect(mockHandleClick).toHaveBeenCalledWith({
+      id: 'BLOCKED',
```
Comment (kevinmamaqi) [inline web/usuaris/src/__tests__/molecules/StatusDropdown.test.tsx:64]: enum

File: web/usuaris/src/__tests__/organisms/FiltersWidget.test.tsx [hunk 4]
```diff
@@ -69,13 +73,83 @@ describe('FiltersWidget', () => {
...
+    render(<FiltersWidget filters={{}} handleFilters={mockHandleFilters} />)
+
+    await waitFor(() =>
+      expect(screen.getByText(/especialitat/i)).toBeInTheDocument()
+    )
+    expect(screen.getByText(/estat/i)).toBeInTheDocument()
+
+    const allDropdowns = screen.getAllByTestId('dropdown-header')
+
+    const dropdownHeader = allDropdowns[1]
```
Comment (kevinmamaqi) [inline web/usuaris/src/__tests__/organisms/FiltersWidget.test.tsx:95]: Selectores especificos

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: web/usuaris/src/components/organisms/UsersTable/UsersTable.tsx [hunk 5]
```diff
@@ -232,8 +225,8 @@ export const UsersTable: FC<TUsersTable> = ({ filtersSelected }) => {
         </ActionsHeader>
       ),
       cell: ({ row }) => {
-        const status: string = row.getValue('status')
-        const id: string = row.getValue('id')
+        const status: UserStatus = row.getValue('status')
+        const id: UserStatus = row.getValue('id')
```
Comment (marmorrei) [inline web/usuaris/src/components/organisms/UsersTable/UsersTable.tsx:229]: Why the id is changed to UserStatus? I think this id type is string, as it was. It is the user id that has to be passed when calling handleStatus to update the user.

Comment (NataliaR-BCN) [reply web/usuaris/src/components/organisms/UsersTable/UsersTable.tsx]: You're right, it is just a mistake copying and pasting!

**Review by kevinmamaqi — APPROVED** [review]

**Review by kevinmamaqi — APPROVED** [review]

**Review by marmorrei — APPROVED** [review]

**Review by mllenas12 — APPROVED** [review]

**Pull request merged**